src/cursor/cur_file.c

```diff
@@ -626,22 +626,26 @@ __curfile_close(WT_CURSOR *cursor)
     WT_ERR(__cursor_copy_release(cursor));
 err:
 
-    /* Only try to cache the cursor if there's no error. */
+    /* Only try to remove the cursor from the cache if there's no error. */
     if (ret == 0) {
         /*
          * If releasing the cursor fails in any way, it will be left in a state that allows it to be
          * normally closed.
          */
         ret = __wt_cursor_cache_release(session, cursor, &released);
-        if (released)
+        if (released) {
```

Comment:
Is it possible to be in the path when closing a bulk cursor? It jumps to the label `done`, might be safer to call `WT_STAT_CONN_DECR_ATOMIC` in there?

Reply:
Looks like we cannot cache special cursors given the following comment:
```
        /*
         * Any cursors that have special configuration cannot be cached. There are some exceptions
         * for configurations that only differ by a cursor flag, which we can patch up if we find a
         * matching cursor.
         */
```